### PR TITLE
4-bit_palette_color

### PR DESCRIPTION
**Pull request by Harvey-Huang**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by Harvey-Huang** [comment]
In #22430(tiff support 4-bit palette), I don't know how to rename the branch name, so I recreate this PR.

File: modules/imgcodecs/src/grfmt_tiff.cpp [hunk 1]
```diff
@@ -312,6 +312,18 @@ bool TiffDecoder::readHeader()
                 result = true;
                 break;
             }
+            case 4:
```
Comment (asmorkalov) [inline modules/imgcodecs/src/grfmt_tiff.cpp:315]: Please add 4-bit support to error message in default branch.

**Review by asmorkalov — APPROVED** [review]
:+1:

File: modules/imgcodecs/src/grfmt_tiff.cpp (hunk 1 above)
Comment (Harvey-Huang) [reply modules/imgcodecs/src/grfmt_tiff.cpp:315]: ok, I've pushed the changed, 4-bit is only supported in palette mode.

**Pull request merged**